frontend/src/routes/guards.js

```diff
@@ -0,0 +1,16 @@
+const addNavigationGuards = (router, store) => {
+  router.beforeEach((to, from, next) => {
+    const publisherId = to.query.pid
+    if (publisherId) {
+      store.commit('publisherId', publisherId)
+      delete to.query.pid
```

Comment:
This deletes the query parameter on next navigation?
Also this could be optimized and remove the `publisherId` constant. But I agree on this implementation. Our task is maintainability & readability, not optimization. Thats task of the compiler/transpiler.